### PR TITLE
SDIT-1444 Roles for Miscellaneous endpoints

### PR DESCRIPTION
**Pull request by steverendell**
Appointments and Booking Resources

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AppointmentsResource.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public class AppointmentsResource {
 
     @ApiResponses({
         @ApiResponse(responseCode = "200", description = "The appointments have been created.")})
+    @HasWriteScope // FIXME: Called by old dps & profile with auth token - needs client token. Write scope allows everyone to call this endpoint.
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/AppointmentsResource.java:42]: This needs attention in a later PR

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AppointmentsResource.java [hunk 2]
```diff
@@ -53,10 +55,10 @@ public List<CreatedAppointmentDetails> createAppointments(
         @ApiResponse(responseCode = "404", description = "The appointment was not found"),
         @ApiResponse(responseCode = "403", description = "The client is not authorised for this operation")
     })
-    @Operation(summary = "Delete an appointment.", description = "Delete appointment.")
+    @Operation(summary = "Delete an appointment.", description = "Requires role GLOBAL_APPOINTMENT and write scope")
     @ResponseStatus(HttpStatus.NO_CONTENT)
+    @PreAuthorize("hasRole('GLOBAL_APPOINTMENT') and hasAuthority('SCOPE_write')")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/AppointmentsResource.java:60]: This PR involves moving annotations from the service layer (no behaviour changes hopefully!)

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 3]
```diff
@@ -159,8 +160,11 @@ public Page<PrisonerBookingSummary> getPrisonerBookingsV2(
...
-
+    public InmateDetail getOffenderBooking(
+        @PathVariable("bookingId") @Parameter(description = "The booking id of offender", required = true) final Long bookingId,
+        @RequestParam(value = "basicInfo", required = false, defaultValue = "false") @Parameter(description = "If set to true then only basic data is returned") final boolean basicInfo,
+        @RequestParam(value = "extraInfo", required = false, defaultValue = "false") @Parameter(description = "Only used when requesting more than basic data, returns identifiers,offences,aliases,sentence dates,convicted status") final boolean extraInfo
+    ) {
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:167]: Formatting - which is in a separate commit

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java [hunk 4]
```diff
@@ -155,7 +155,7 @@ public UserDetail getUserDetails(@PathVariable("username") @Parameter(descriptio
     }
 
     @ApiResponses({@ApiResponse(responseCode = "200", description = "The list of user details")})
-    @Operation(summary = "Returns the user details for supplied usernames - POST version to allow large user lists.", description = "user details for supplied usernames")
+    @Operation(summary = "Returns the user details for supplied usernames - POST version to allow large user lists.", description = "Requires role STAFF_SEARCH")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/UserResource.java:158]: Minor tweaks related to previous PR

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 5]
```diff
@@ -798,8 +891,9 @@ public List<PrisonDetails> getBookingVisitsPrisons(
 
     @ApiResponses({
         @ApiResponse(responseCode = "201", description = "The appointment has been recorded. The updated object is returned including the status.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ScheduledEvent.class))})})
-    @Operation(summary = "Create appointment for offender.", description = "Create appointment for offender.")
+    @Operation(summary = "Create appointment for offender.", description = "Requires role VIEW_PRISONER_DATA or GLOBAL_SEARCH, or booking is in caseload")
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:894]: No `VIEW_PRISONER_DATA` on the `@VerifyBookingAccess`.

**Review by mikehalmamoj — APPROVED** [review]

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**